avocado/plugins/nlist.py

```diff
@@ -46,6 +52,15 @@ def run(self, config):
         matrix, stats, tag_stats, resolution_matrix = self._get_resolution_matrix(config,
                                                                                   resolutions)
         self._display(matrix, stats, tag_stats, resolution_matrix, config.get('verbose'))
+        recipes_directory = config.get('write_recipes_to_directory')
+        if recipes_directory is not None:
+            fmt = '%%0%uu.json' % len(str(len(matrix)))
+            index = 1
+            for resolution in resolutions:
+                if resolution.result == resolver.ReferenceResolutionResult.SUCCESS:
+                    for res in resolution.resolutions:
+                        res.write_json(os.path.join(recipes_directory, fmt % index))
+                        index += 1
```

Comment:
Can't we elaborate on a more friendly file naming? Maybe the test name or the file name with the test name as a suffix.

Reply:
I though about a few options:

1) Using the URI
2) Using a sequence + URI
3) Using a sequence + kind + URI
4) Allowing the user to supply a format string

But, to postpone the discussion, since I don't think it's critical to the implementation of the features that I intended to add on top of this,I though it'd be better to postpone the discussion and keep the naming as simple as possible.

BTW, I think it's worth mentioning where I'm planning to take this:

1) reproducible executions: think of the developer/user/QE that finds a bug, and wants someone else to reproduce that one code.  Right now, Runnable recipes describe part of what is necessary.  This will be augmented with more metadata (https://trello.com/c/lumR8u8Y/1526-rfc-nrunner-extended-metadata) and with Tasks recipes, that will contain more information on how that Runnable is supposed to be executed (say parameters given to the job and consequently tests).

2) pausing / resuming jobs: if the job input (the recipes) and its status is serialized in the filesystem, it should be possible to have commands that tell it to pause spawning new tasks, resume spawning new tasks, or even change the rate (depending on the system load?).

So again, a lot of it is experimental... and that's why I'm trying to keep the base functionality to a bare minimum.

Reply:
Okay, thinking about the main functionality, the naming mechanism can be added later.